swablu/web.py

```diff
@@ -403,12 +404,20 @@ async def post(self, hack_id):
                 break
         if key != hack_id or not hack:
             return self.redirect('/edit')
-        hack['name'] = self.get_body_argument('name', '')
-        hack['description'] = self.get_body_argument('description', '')
-        hack['hack_type'] = self.get_body_argument('hack_type', '')
+
+        editing = bool(hack['message_id'])
+
+        try:
+            hack['name'] = self.get_body_argument('name')
+            hack['description'] = self.get_body_argument('description')
+            hack['hack_type'] = self.get_body_argument('hack_type')
+        except MissingArgumentError:
+            return self.redirect(f'/edit/{hack_id}?missing_arg=1')
```

Comment:
Could the name of the missing argument be included in the redirect url?

Reply:
For what purpose? Displaying which argument was missing to the user? What if there's more than one? (Keep in mind that the exception is triggered as soon as one argument is missing, so we don't have the full list if there's multiple).